### DISCUSSION TITLE
WithCancel/WithTimeout contexts not canceling DB query

### DISCUSSION THREAD

**Discussion by gcemaj** [uptrace/bun#792] [Q&A]
When using `ctx.WithTimeout` or `ctx.WithCancel` the call on the golang side gets cancelled but i still see the long running query in the Postgres side.

I would hope/want/expect the call to be cancelled as well in the DB side?

Is there away to do that?

**Comment by gcemaj** [comment]
Alternatively, is there a way to set the stament_timeout for the connection using the pgdriver dialect?

**Comment by vmihailenco** [comment]
You probably can use `pgdriver.WithConnParams` to set statement timeout.

**Comment by maximerety** [comment]
Out of curiosity, are there any plans to support query cancelation when the context is canceled or timeouts?

A feature that would be similar to the context cancelation implemented in [lib/pq](https://github.com/lib/pq/blob/8446d16b8935fdf2b5c0fe333538ac395e3e1e4b/conn_go18.go#L142) via the sending of the `80877102` cancel request code (see: https://www.postgresql.org/docs/15/protocol-message-formats.html#PROTOCOL-MESSAGE-FORMATS-CANCELREQUEST).

Thanks.

**Comment by phoenix2x** [comment]
Moreover, this feature was supported by go-pg: https://github.com/go-pg/pg/blob/f7e1c98fed3044ae4e195a0e063f69185fb1885b/base.go#L529

I'm not convinced that the feature is always useful, especially for short queries, as it might be more efficient to just let the query finish instead of [opening the new connection and sending the cancel signal](https://www.postgresql.org/docs/14/protocol-flow.html#id-1.10.5.7.10). So really want to hear if it was deliberately omitted or ...?


Thank you.
